base/array.jl

```diff
@@ -664,14 +673,11 @@ function collect_to!(dest::AbstractArray{T}, itr, offs, st) where T
         y = iterate(itr, st)
         y === nothing && break
         el, st = y
-        if el isa T || typeof(el) === T
```

Comment:
Last time we discussed this, it was needed to get the best generated code. Unless you can find that old thread and check that it's no longer needed, I'd keep it as-is.

Reply:
Ok I added it back in